src/main/java/lotto/controller/LottoController.java

```diff
@@ -0,0 +1,80 @@
+package lotto.controller;
+
+import lotto.converter.LottoWinningNumbersConverter;
+import lotto.domain.LottoWinningNumbers;
+import lotto.domain.Lottos;
+import lotto.dto.LottoOutputDto;
+import lotto.service.LottoBuyService;
+import lotto.service.LottoCheckService;
+import lotto.validator.BonusNumberValidator;
+import lotto.validator.PurchaseAmountValidator;
+import lotto.validator.WinningNumberValidator;
+import lotto.view.InputView;
+import lotto.view.OutputView;
+
+public class LottoController {
+
+    private final InputView inputView;
+    private final OutputView outputView;
+    private final LottoBuyService lottoBuyService;
+    private final LottoCheckService lottoCheckService;
+
+    public LottoController(InputView inputView, OutputView outputView, LottoBuyService lottoBuyService,
+                           LottoCheckService lottoCheckService) {
+        this.inputView = inputView;
+        this.outputView = outputView;
+        this.lottoBuyService = lottoBuyService;
+        this.lottoCheckService = lottoCheckService;
+    }
+
+    public void buyLotto() {
+        String purchaseAmount = this.enterPurchaseAmount();
+
+        Lottos lottos = lottoBuyService.buyLotto(purchaseAmount);
+        outputView.showLottos(lottos);
+
+        String winningNumbers = this.enterWinningNumber();
+        String bonusNumber = this.enterBonusNumber(winningNumbers);
+
+        LottoWinningNumbers lottoWinningNumbers = LottoWinningNumbersConverter.toLottoWinningNumbers(winningNumbers,
+                bonusNumber);
+        LottoOutputDto lottoOutputDto = lottoCheckService.checkLottos(purchaseAmount, lottoWinningNumbers, lottos);
+        outputView.showResult(lottoOutputDto);
+    }
+
+    private String enterPurchaseAmount() {
+        while (true) {
+            try {
+                String purchaseAmount = inputView.enterPurchaseAmount();
+                PurchaseAmountValidator.validatePurchaseAmount(purchaseAmount);
+                return purchaseAmount;
+            } catch (IllegalArgumentException e) {
+                outputView.showErrorMessage(e.getMessage());
+            }
+        }
+    }
```

Comment:
Supplier<T> 함수형 인터페이스를 사용하면 반복되는 while-try-catch문을 제거해볼 수 있겠습니다

```suggestion
    private String enterPurchaseAmount() {
        return inputWithRetry(() -> {
            String purchaseAmount = inputView.enterPurchaseAmount();
            PurchaseAmountValidator.validatePurchaseAmount(purchaseAmount);
            return purchaseAmount;
        });
    }
    
    private <T> T inputWithRetry(Supplier<T> supplier) {
        while (true) {
            try {
                return supplier.get();
            } catch (IllegalArgumentException e) {
                outputView.showErrorMessage(e.getMessage());
            }
        }
    }
```

Reply:
몰랐던 부분인데 새롭게 배워갑니다.. 감사합니다